andela/mrm_api
# #166054173 Update user role to admin

#### What does this PR do?
Currently, there's an existing implementation for changing a user's role, there is also an implementation for sending an email notification to a user when his role is being changed to admin. The aim of this PR is to give that user the privilege of accepting to become an admin by clicking on a button that comes with the email notification.
#### Description of Task to be completed?
- Send email notification to newly added admin
- implement mutation to update user role to admin
- refactor the previous implementation
- write tests for functionality
- Ensure existing tests pass.
#### How should this be manually tested?
- Pull this branch ch-activate-admin-onclick-email-button-166054173
- Make sure your application containers are running
- Navigate to this [PR](https://github.com/andela/mrm_api/pull/388) to run the first mutation that sends the email notification to the user
- After running that mutation, Run this mutation
```
mutation {
  updateUserRoleToAdmin(email: "email@andela.com") {
    user {
      name
      roles{
        role
      }
    }
  }
}
```
- For the `email` field in the mutation, make use of an existing valid email address in your User model.
#### What are the relevant pivotal tracker stories?
[166054173](https://www.pivotaltracker.com/story/show/166054173)
#### Screenshots (if appropriate)
<img width="1195" alt="Screen Shot 2019-06-03 at 9 01 52 AM" src="https://user-images.githubusercontent.com/42177767/58786013-9aed7d00-85de-11e9-9370-914d07dab795.png">
